### ISSUE TITLE
Upgrading form 4.2.3 -> 4.3.0 - synchthing crashes and corrupts os. 

### ISSUE THREAD

**Issue by blakekl** [XK9274/syncthing-app-miyoo#15]
After upgrading, syncthing does not appear to be running. My main syncthing computer shows the MM+ as never connected anymore, files aren't syncing. when I try to run the syncthing script on the device, I see the message that it has not been configure, then it shows that the IP address URL to set it up, then crashes. After this crash, I can no longer use any of the other Onion OS apps. My games are also all missing. The whole OS becomes unusable. I have to redo the upgrade process to get my OS functioning again.

**Comment by XK9274** [comment]
Hey.

Granted syncthing can hammer the CPU/SD if you opt to backup your full SD instead of specific directories (Saves), I don't think Syncthing or Onion has caused any issues here. 

Sounds like your SD card has just gone corrupt.

What SD is it?

**Comment by blakekl** [comment]
@XK9274 

No, it's not the SD card. I can run everything again just fine if I repeat the update process. I don't sync the entire file system. I only sync battery saves, save states, and a folder to sync some game files to make transferring things over easier. My SD card is a sandisk ultra pro 32gb, so it's not overly large or anything either. Everything worked great before the update. I could probably just delete the entire config file and redo all the setup, but I figured there must be a problem here and I would like to try to preserve the state to debug if you'd like to investigate it some more. I tried to backup the whole config file to my windows machine and it does complain that there is some log file that is corrupted and can't be read / copied over. I don't know if it's related to that.

**Comment by Splozy** [comment]
Similar issue here. Syncthing logs no connection activity from the Miyuu or connection refused. Tried powering device off and running/killing the app a few times.

**Comment by XK9274** [comment]
Is Syncthing even running? Have you tried to backup your config dir and reinstall the full app?

**Comment by Pointy-Bits** [comment]
I was upgrading from 4.2 (relunctantly) to 4.3, backed up my app folders hoping to just drop them into the upgraded install, and syncthing doesn't work, and interferes with the WiFi menu (i removed BetterWifi and did a clean install of onionOS and this is still happening). I'm not sure what else I can do but SyncThing was great when it was working 👍 

Nice work on your apps!

**Comment by Pointy-Bits** [comment]
So digging around a little more, I was able to get syncthing up and broadcasting on localhost:8384, with what looks to be my previous configuration, but with this error.
![miyoo_syncthing](https://github.com/XK9274/syncthing-app-miyoo/assets/107223856/6fcda46b-56f5-4a06-b38c-09837656a0ea)

**Comment by Pointy-Bits** [comment]
*update*
1 .enabled SSH
2. navigated to /mnt/SDCARD/Saves/CurrentProfile/saves
3. mkdir .stfolder (syncthing didn't like the existing folder)
4. rescanned, uploaded, and at least for now, syncthing is working

Whether it plays nice with the rest of apps, I have yet to find out (syncthing really needs to run after the miyoo grabs the correct time from the internet)

sorry for spam

**Comment by blakekl** [comment]
This has happened to me again going from 4.3.0 -> 4.3.1-1

Where are the syncthing config files located? Want to delete them and start over again.

**Comment by XK9274** [comment]
A lot has changed since this issue, closing.

See latest release. 

https://github.com/XK9274/syncthing-app-miyoo/releases/tag/v1.0.0
